### ISSUE TITLE
Function.call_sites might not be complete if the function has not been navigated to

### ISSUE THREAD

**Issue by JavierYuste** [Vector35/binaryninja-api#1744]
I noticed that the variable call_sites of a given function is updated when navigating into such function (after all analysis is already finished). So if I issue the following line in the Python console, I get 9 references:

`current_view.get_function_at(0xNNNN).call_sites`

 However, if I navigate to the function at 0xNNNN and then issue the same line, I get 10 references. The one that was missing is a call to an imported function located at the IAT. Is that expected? Do I have to update the lists in the background in order to get all references from each function as if I had navigated through them?

**Comment by JavierYuste** [comment]
Sure! I will attach the file here (password is "infected"). Quick test is as follows:

1. `current_view.get_function_at(0x410b0b).call_sites` -> Gets X addresses
2. Navigate to the function at 0x410b0b in the UI
3. `current_view.get_function_at(0x410b0b).call_sites` -> Gets X+1 addresses, the call to SetErrorMode at 0x410b12

However, I am afraid I am not being able to reproduce it today. I noticed the issue on May 31, at around 10 PM (CEST), using an updated version of BN from the dev channel. I have updated BN since then, which may have solved the issue.

[svchost.bin.zip](https://github.com/Vector35/binaryninja-api/files/4725673/svchost.bin.zip)

**Comment by plafosse** [comment]
I just tested this and it seems to be fixed :) We somewhat recently made a change to how analysis is updated that probably fixed this issue.

```
>>> sites = set([r.address for r in current_view.get_function_at(0x410b0b).call_sites])
>>> here = 0x410b0b
>>> sites2 = set([r.address for r in current_view.get_function_at(0x410b0b).call_sites])
>>> sites-sites2
set()
```
Should be fixed as of build 2.3.2713
